x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackPlugin.java

```diff
@@ -225,14 +232,16 @@ public Settings additionalSettings() {
                                                NodeEnvironment nodeEnvironment, NamedWriteableRegistry namedWriteableRegistry) {
         List<Object> components = new ArrayList<>();
 
+        final SSLService sslService = new SSLService(environment);
+        setSslService(sslService);
```

Comment:
nit: this looks like the only use of setSslService, can it be removed and just set the setonce member directly?

Reply:
`LocalStateCompositeXPackPlugin` does crazy stuff.
This is the only use of the method, but we have multiple implementations :(